src/syntax_analysis.rs

```diff
@@ -0,0 +1,192 @@
+use std::collections::HashMap;
+use syntect::parsing::syntax_definition::{
+    ContextReference, MatchOperation, MatchPattern, Pattern, SyntaxDefinition,
+};
+use syntect::parsing::{Scope, SyntaxSet, SyntaxSetBuilder};
+
+/// Used to look up (by name) what dependencies a given [SyntaxDefinition] has
+type SyntaxToDependencies = HashMap<String, Vec<Dependency>>;
+
+/// Used to look up what [SyntaxDefinition] that corresponds to a given [Dependency]
+type DependencyToSyntax<'a> = HashMap<Dependency, &'a SyntaxDefinition>;
+
+/// Represents a dependency on an external `.sublime-syntax` file.
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+enum Dependency {
+    /// By name. Example YAML: `include: C.sublime-syntax`
+    ByName(String),
+
+    /// By scope. Example YAML: `embed: scope:source.c`
+    ByScope(Scope),
+}
+
+/// Change to true to make syntax dependency analysis print more details of what it is seeing
+const VERBOSE: bool = false;
+
+/// Generates independent [SyntaxSet]s after analyzing dependencies between syntaxes
+/// in a [SyntaxSetBuilder], and then prints the reults.
+pub(crate) fn print_syntax_dependencies(syntax_set_builder: &SyntaxSetBuilder) {
+    println!("Constructing independent SyntaxSets...");
+    let independent_syntax_sets = build_independent_syntax_sets(syntax_set_builder);
+
+    println!("Independent SyntaxSets:");
+    for syntax_set in independent_syntax_sets {
+        let names = syntax_set
+            .syntaxes()
+            .iter()
+            .map(|syntax| &syntax.name)
+            .collect::<Vec<&String>>();
+        println!("{:?}", names);
```

Comment:
Another nitpicky comment. Using the debug output for this sort of output is completely fine, but we typically want to avoid using `:?`. Here you could use
```rs
println!("{}", names.join(", "));
```
instead.

Reply:
Hmm, using `.join` here just adds unnecessary complexity I think. Like you say, the output is not meant to be consumed by "normal" users. So I would prefer to keep it as it is.

If you insist, I will of course change it though :)

Reply:
no, please leave the debug output in that case.